tests/integration/models/applicationInvite.spec.ts

```diff
@@ -0,0 +1,132 @@
+
+describe('Application Invite Model', function () {
+	givenLoggedInUser(before);
+
+	describe('given no application invite', function () {
+		givenAnApplication(before);
+		describe('balena.models.application.invite.getAllByApplication()', function () {
+			it('shoud return an empty Array', function () {
+				const promise = balena.models.application.invite.getAllByApplication(1);
+				return expect(promise).to.become([]);
+			});
+
+			it('should support a callback with no options', function (done) {
+				(balena.models.application.invite.getAllByApplication as (
+					...args: any[]
+				) => any)(this.application.id, function (
+					_err: Error,
+					applicationInvite: BalenaSdk.ApplicationInvite[],
+				) {
+					expect(applicationInvite).to.deep.equal([]);
+					return done();
+				});
+			});
+		});
+
+		describe('balena.models.application.invite.create()', function () {
+			it('should create and return an application invite', async function () {
+				const response = await balena.models.application.invite.create(
+					this.application.id,
+					{
+						collaborator: TEST_EMAIL,
+						roleName: TEST_ROLE,
+						message: TEST_MESSAGE,
+					},
+				);
+				expect(response).to.have.property('id');
+				const invites = await balena.models.application.invite.getAllByApplication(
+					this.application.id,
+					{
+						$expand: {
+							is_invited_to__application: { $select: ['id'] },
+							invitee: { $select: ['email'] },
+							application_membership_role: { $select: ['name'] },
+						},
+						$select: ['message'],
+					},
+				);
+				expect(invites).to.have.length(1);
+				expect(invites[0].is_invited_to__application[0].id).to.equal(
+					this.application.id,
+				);
+				expect(invites[0].application_membership_role[0].name).to.equal(
+					TEST_ROLE,
+				);
+				expect(invites[0].message).to.equal(TEST_MESSAGE);
+			});
+		});
+	});
+
+	describe('given a single application invite [contained scenario]', function () {
+		givenAnApplication(before);
+		before(async function () {
+			const applicationInvite = await balena.models.application.invite.create(
+				this.application.id,
+				{
+					collaborator: TEST_EMAIL,
+					roleName: TEST_ROLE,
+					message: TEST_MESSAGE,
+				},
+			);
+			return (this.applicationInvite = applicationInvite);
+		});
+
+		describe('balena.models.application.invite.getAllApplication()', () => {
+			it('should become the list of application invites', async function () {
+				const applicationInvites = await balena.models.application.invite.getAllByApplication(
+					this.application.id,
+				);
+				expect(applicationInvites).to.have.length(1);
+			});
+
+			it('should support arbitrary pinejs options', async function () {
+				const applicationInvites = await balena.models.application.invite.getAllByApplication(
+					this.application.id,
+					{
+						$expand: { invitee: { $select: ['email'] } },
+					},
+				);
+				expect(applicationInvites[0].invitee[0].email).to.equal(TEST_EMAIL);
```

Comment:
You should be able to do:
```suggestion
				assertDeepMatchAndLength(applicationInvites, [applicationInvites: [{email: TEST_EMAIL}]]
```

You will need to add the following import:
```
const { assertDeepMatchAndLength } = require('../../util');
```

Reply:
Actually while trying to help avoiding the `any` import here, I've opened https://github.com/balena-io/balena-sdk/pull/904.
After that you should be able to use a normal
```
import { assertDeepMatchAndLength } from '../../util';
```